### DISCUSSION TITLE
Session Invalidation errors in Kubernetes pods that use Redis for session management.

### DISCUSSION THREAD

**Discussion by rahil-pandya** [redisson/redisson#6010] [Q&A]
We've been noticing that users are intermittently getting kicked out of their sessions, and upon investigation, we find 

> java.lang.IllegalStateException: setAttribute: Session {sessionid} has already been invalidated

 errors in our logs. This seems to indicate issues with session management. We use Kubernetes to orchestrate multiple pods, each running a Tomcat server, while using Redis as the session storage.

Could anyone shed light on common causes for sessions becoming invalidated unexpectedly? Are there known patterns or configurations that might contribute to this issue? 

Thank you!

**Comment by mrniko** [comment]
Multiple threads or requests are attempting to modify the same session object concurrently. Once a session is invalidated, any subsequent attempt to modify it results in this error.
